extensions/wikia/Search/classes/Result/ResultHelper.php

```diff
@@ -59,7 +61,11 @@ public static function extendResult( $result, $pos, $descWordLimit, $imageSizes,
 		if ( $query ) {
 			$lang = $wikiaSearchHelper->getLangForSearchResults();
 			$centralUrl = $wikiaSearchHelper->getCentralUrlFromGlobalTitle( $lang );
-			$globalSearchUrl = $wikiaSearchHelper->getGlobalSearchUrl( $centralUrl ) . '?search=' . urlencode( $query );
+			$fulltext = $wgUser->getGlobalPreference( 'enableGoSearch' ) ? 0 : 'Search';
+			$globalSearchUrl = $wikiaSearchHelper->getGlobalSearchUrl( $centralUrl ) .
+				'?search=' . urlencode( $query ) .
+				'&fulltext=' . urlencode( $fulltext ).
+				'&resultsLang=' . urlencode( $lang );
```

Comment:
Maybe we could use http://php.net/manual/en/function.http-build-query.php or https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L376-L376